src/main/java/teammates/storage/search/SearchManager.java

```diff
@@ -85,6 +87,69 @@ public static void putDocument(String indexName, Document document) {
     }
 
     /**
+     * Batch creates or updates the search documents for the given documents and index.
+     */
+    public static void putDocuments(String indexName, List<Document> documents) {
+        Index index = getIndex(indexName);
+
+        int delay = 2;
+        for (int attempts = 0; attempts < MAX_RETRIES; attempts++) {
+            try {
+                List<Document> documentsToRetry = tryPuttingDocuments(index, documents);
+                boolean isSuccessful = documentsToRetry.isEmpty();
+
+                if (Config.PERSISTENCE_CHECK_DURATION == 0) {
+                    continue;
+                }
+
+                int elapsedTime = 0;
+                while (!isSuccessful && elapsedTime < Config.PERSISTENCE_CHECK_DURATION) {
+                    ThreadHelper.waitBriefly();
+                    isSuccessful = tryPuttingDocuments(index, documentsToRetry).isEmpty();
+
+                    // check before incrementing to avoid boundary case problem
+                    if (!isSuccessful) {
+                        elapsedTime += ThreadHelper.WAIT_DURATION;
+                    }
+                }
+                if (elapsedTime >= Config.PERSISTENCE_CHECK_DURATION) {
+                    log.info(String.format(ERROR_EXCEED_DURATION, documents, indexName));
+                }
+
+            } catch (PutException e) {
+                if (StatusCode.TRANSIENT_ERROR.equals(e.getOperationResult().getCode())) {
+                    // if it's a transient error in the server, it can be retried
+                    ThreadHelper.waitFor(delay * 1000);
+                    delay *= 2; // use exponential backoff
+                    continue;
+                } else {
+                    log.severe(String.format(ERROR_NON_TRANSIENT_BACKEND_ISSUE, documents, indexName)
+                            + TeammatesException.toStringWithStackTrace(e));
+                    break;
+                }
+            }
+        }
+    }
+
+    /**
+     * Puts documents in index.
```

Comment:
Just the ones already inside the diff. No need to make the diff bigger just of fix cosmetic problems.